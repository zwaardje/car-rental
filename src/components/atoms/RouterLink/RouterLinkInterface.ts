import { ReactNode, HTMLAttributes } from 'react';

export interface RouterLink extends HTMLAttributes<any>{
    url: string;
    children: ReactNode | string;
    state?: any;
    search?: any;
}
