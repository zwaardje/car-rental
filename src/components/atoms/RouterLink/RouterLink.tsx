import * as React from 'react';
import { Link } from 'react-router-dom';

import { RouterLink as RouterLinkInterface } from './RouterLinkInterface';

export interface Props extends RouterLinkInterface {}
export interface State {}

export default class RouterLink extends React.Component<Props, State> {
    static displayName = 'RouterLink';

    render() {
        const { children, url, state, search, className } = this.props;

        return (
            <Link className={`link ${className ? ` ${className}` : ''}`} to={{
                state,
                search,
                pathname: url,
            }}>
                {children}
            </Link>
        );
    }
}
'';
