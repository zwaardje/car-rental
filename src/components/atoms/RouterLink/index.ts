import RouterLink from './RouterLink';
export default RouterLink;
