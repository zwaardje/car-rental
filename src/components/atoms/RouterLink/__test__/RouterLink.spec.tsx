import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import RouterLink from '../RouterLink';

import { defaultProps } from '../__data__/RouterLink.mockdata';

describe('RouterLink', () => {

    it('renders primary routerlink correctly', () => {
        const props = { ... defaultProps };
        const tree = renderer.create(
            <Router>
                <RouterLink {... props} />
            </Router>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
