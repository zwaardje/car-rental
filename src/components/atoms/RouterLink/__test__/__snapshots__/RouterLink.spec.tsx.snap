// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouterLink renders primary routerlink correctly 1`] = `
<a
  className="link link__primary"
  href="#"
  onClick={[Function]}
/>
`;

exports[`RouterLink renders secondary routerlink correctly 1`] = `
<a
  className="link link__secondary"
  href="#"
  onClick={[Function]}
/>
`;

exports[`RouterLink renders underline routerlink correctly 1`] = `
<a
  className="link link__underline"
  href="#"
  onClick={[Function]}
/>
`;
